### PR TITLE
broadcast_to function

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `broadcast_to` function.
related to #642

### CODE REVIEW

File: tests/chainer_tests/functions_tests/array_tests/test_broadcast.py [hunk 1]
```diff
@@ -102,4 +102,67 @@ def test_no_args(self):
             functions.broadcast()
 
 
+@testing.parameterize(
+    {'in_shape': (3, 1, 5), 'out_shape': (3, 2, 5)},
+    {'in_shape': (5,),      'out_shape': (3, 2, 5)},
+    {'in_shape': (5,),      'out_shape': (3, 2, 5)},
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/array_tests/test_broadcast.py]: Duplicate test?

File: chainer/functions/array/broadcast.py [hunk 2]
```diff
@@ -61,3 +61,56 @@ def broadcast(*args):
...
+        if hasattr(xp, 'broadcast_to'):
+            return xp.broadcast_to(x, self._shape),
+        else:
+            # numpy 1.9 doesn't support broadcast_to method
+            dummy = xp.empty(self._shape)
+            bx, _ = xp.broadcast_arrays(x, dummy)
+            return bx,
+
+    def backward(self, xs, grads):
+        return tuple(_backward_one(x, g) for x, g in six.moves.zip(xs, grads))
```
Comment (okuta) [inline chainer/functions/array/broadcast.py]: How about `return _backward_one(xs[0], grads[0]),`?

File: chainer/functions/array/broadcast.py [hunk 3]
```diff
@@ -61,3 +61,56 @@ def broadcast(*args):
...
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+
+        ndim = type_check.Variable(len(self._shape), 'len(shape)')
+        type_check.expect(in_types[0].ndim <= ndim)
+
+        shape = in_types[0].shape.eval()
+        for i in range(len(shape)):
+            j = -i - 1
```
Comment (okuta) [inline chainer/functions/array/broadcast.py]: Why use `j`?

File: chainer/functions/array/broadcast.py [hunk 4]
```diff
@@ -61,3 +61,56 @@ def broadcast(*args):
...
     """
     return Broadcast()(*args)
+
+
+class BroadcastTo(function.Function):
+
+    """Function that broadcasts an array to a new shape."""
+
+    def __init__(self, shape):
+        shape = tuple(shape)
```
Comment (okuta) [inline chainer/functions/array/broadcast.py]: How about `self._shape = tuple(shape)` ?

File: tests/chainer_tests/functions_tests/array_tests/test_broadcast.py (hunk 1 above)
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/array_tests/test_broadcast.py]: oops

File: chainer/functions/array/broadcast.py (hunk 3 above)
Comment (unnonouno) [inline chainer/functions/array/broadcast.py]: for readability

File: chainer/functions/array/broadcast.py [hunk 5]
```diff
@@ -61,3 +61,55 @@ def broadcast(*args):
...
+        self._shape = tuple(shape)
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+
+        ndim = type_check.Variable(len(self._shape), 'len(shape)')
+        type_check.expect(in_types[0].ndim <= ndim)
+
+        shape = in_types[0].shape.eval()
+        for i in range(len(shape)):
```
Comment (okuta) [inline chainer/functions/array/broadcast.py]: How about `for i in range(-1, -len(shape), -1)`?

**Comment by okuta** [comment]
LGTM!

**Pull request merged**